docs(repo): add missing rules

heitortsergent opened the pull request:
Fixes #2193.

Adds the following 3 rules that were missing from the OpenAPI Rules doc:

- oas2-valid-media-example
- oas3-valid-media-example
- oas3-valid-schema-example

**Checklist**

- [ ] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

heitortsergent committed:
docs(repo): reorder oas2-operation-formData-consume-check rule

heitortsergent committed:
docs(repo): add oas2-valid-media-example rule example

heitortsergent committed:
docs(repo): add oas3-valid-media-example rule

heitortsergent committed:
docs(repo): run prettier

heitortsergent committed:
docs(repo): add oas3-valid-schema-example rule

heitortsergent committed:
docs(repo): update pet examples

heitortsergent committed:
docs(repo): add more details to new oas3 examples descriptions

heitortsergent committed:
docs(repo): run prettier

heitortsergent committed:
docs(repo): update links for markdown-link-checker

heitortsergent committed:
docs(repo): fix link for markdown-link-checker 2

heitortsergent commented:
Questions I have for this PR:

- Are the examples I have added for the rules correct/ok?
- Are the 3 rules **Recommended** supposed to be "yes"? I marked them as "yes" by looking at [rulesets/src/oas/index.ts](https://github.com/stoplightio/spectral/blob/develop/packages/rulesets/src/oas/index.ts#L659).
- Can I reorder the rules in the OpenAPI Rules page so they're in alphabetical order?

philsturgeon commented:
Yes, yes, and yes!

philsturgeon committed:
Merge branch 'develop' into docs/missing-rules

P0lip commented on docs/reference/openapi-rules.md:
```diff
@@ -658,3 +727,122 @@ Validate structure of OpenAPI v3 specification.
+
+### oas3-valid-media-example
+
+Examples must be valid against their defined schema. This rule is applied to Media objects, such as requests and responses.
```
request and responses aren't media objects though.
Media Type Object is just a standalone object https://spec.openapis.org/oas/v3.1.0#media-type-object that may be present under "content" in requests or responses

philsturgeon replied:
I think this is correct in the sense that a media type object is used in a request and a response. Requests and responses both have media types in there somewhere.

P0lip reviewed (changes requested):
Looks good apart from that one comment

heitortsergent committed:
docs(repo): update oas3-valid-media-example description

heitortsergent committed:
Merge branch 'develop' into docs/missing-rules

heitortsergent commented:
@P0lip thanks for the review! I updated the description and removed the request/response mention.

Any thoughts on if I can reorder the rules alphabetically? 🤓

philsturgeon commented:
> Any thoughts on if I can reorder the rules alphabetically? 🤓 

Absolutely.

heitortsergent committed:
docs(repo): reorder openapi rules alphabetically
